### PR TITLE
MINOR: improve docstrings on SessionContext

### PR DESCRIPTION
**Pull request by alamb**
# Which issue does this PR close?

N/A

 # Rationale for this change
I started making some minor changes I noticed while reviewing https://github.com/apache/arrow-datafusion/pull/3600 (see  https://github.com/apache/arrow-datafusion/pull/3600#discussion_r979231669) 

# What changes are included in this PR?
Improve docstrings, fix some links

# Are there any user-facing changes?
Better API docs

### CODE REVIEW

File: datafusion/core/src/physical_plan/aggregates/mod.rs [hunk 1]
```diff
@@ -90,8 +90,8 @@ pub struct PhysicalGroupBy {
     null_expr: Vec<(Arc<dyn PhysicalExpr>, String)>,
     /// Null mask for each group in this grouping set. Each group is
     /// composed of either one of the group expressions in expr or a null
-    /// expression in null_expr. If groups[i][j] is true, then the the
-    /// j-th expression in the i-th group is NULL, otherwise it is expr[j].
+    /// expression in null_expr. If `groups[i][j]` is true, then the the
```
Comment (alamb) [inline datafusion/core/src/physical_plan/aggregates/mod.rs:93]: this is a driveby cleanup -- `rustdoc` was complaining that `i` and `j` were not valid links 😆

**Review by tustvold — APPROVED** [review]
